src/Plugins/Reminder.php

```diff
@@ -0,0 +1,357 @@
+                return $this->chatClient->postReply($command, "I'm sorry Dave, I'm afraid I can't do that");
+            }
+
+            $reminders = yield $this->storage->getKeys($command->getRoom());
+            if ($reminders !== []) {
+                foreach ($reminders as $key){
+                    yield $this->storage->unset($key, $command->getRoom());
+                }
+            }
+            return $this->chatClient->postMessage($command->getRoom(), "Reminders are gone.");
+        });
+    }
+
+    private function unset(Command $command) {
+        $messageId = $command->getParameter(1) ?? false;
+
+        return resolve(function() use($command, $messageId) {
+            if (!yield $this->admin->isAdmin($command->getRoom(), $command->getUserId())) {
+                return $this->chatClient->postReply($command, "I'm sorry Dave, I'm afraid I can't do that");
+            }
+
+            if(!$messageId){ return $this->chatClient->postMessage($command->getRoom(), self::USAGE); }
+            $key = $messageId;
+
+            if (yield $this->storage->exists($key, $command->getRoom())) {
+                yield $this->storage->unset($key, $command->getRoom());
+                return $this->chatClient->postMessage($command->getRoom(), "Reminder unset.");
+            }
+            return $this->chatClient->postReply($command, "I'm sorry, I couldn't find that key.");
+        });
+    }
+
+    private function setReminder(Command $command): Promise
+    {
+        $commandName = $command->getCommandName();
+        return resolve(function() use($command, $commandName) {
+
+            switch ($commandName){
+                case 'in':
+                    if(preg_match(self::TIME_STRING_REGEX, implode(" ", $command->getParameters()), $matches)){
+                        $time = $matches['time'] ?? false;
+                        $text = $matches['string'] ?? false;
+                    }
+                    break;
+                case 'at':
+                    $time = $command->getParameter(0) ?? false; // 24hrs
+                    if($time && preg_match(self::TIME_FORMAT_REGEX, $time)){ // !!at 16:00 remind | !!at monday next week remind?
+                        $text = implode(" ", array_diff($command->getParameters(), array($time)));
+                    }
+                    break;
+                case 'reminder':case 'remind':
+                    $parameters = implode(" ", $command->getParameters());
+                    if(!preg_match(self::REMINDER_REGEX, $parameters, $matches)){
+                        return $this->chatClient->postMessage($command->getRoom(), self::USAGE);
+                    }
+
+                    $time = $matches[2] ?? false;
+                    $text = $matches[1] ?? false;
+                    break;
+            }
+
+            if(!isset($time) || !$time): return $this->chatClient->postMessage($command->getRoom(), "Have a look at the time again, yo!"); endif;
```

Comment:
Either do 

    if(!isset($time) || !$time) return $this->chatClient->postMessage($command->getRoom(), "Have a look at the time again, yo!");

or

    if(!isset($time) || !$time) {
        return $this->chatClient->postMessage($command->getRoom(), "Have a look at the time again, yo!");
    }

instead of alternative `if` statement on a single line